crypto/src/main/kotlin/net/corda/v5/crypto/DigitalSignature.kt

```diff
@@ -36,4 +36,14 @@ open class DigitalSignature(
          */
         val context: Map<String, String>
     ) : DigitalSignature(bytes)
+
+    class WithId(
+        /**
+         * SHA-256 digest of the public key. TODO add reasoning for hashing public hey
```

Comment:
To be added. Suggest "SHA-256 digest of the encoded public key. We propagate hash rather than public key itself since these values should only be used for comparison. The actual public keys need to be propagated by some other mechanism before the signature can be validated."